projects/rocprofiler-systems/source/lib/rocprof-sys/library/components/pthread_create_gotcha.cpp

```diff
@@ -540,11 +542,38 @@ pthread_create_gotcha::get_native_handles()
     return _v;
 }
 
+namespace
+{
+bool
+is_rocm_internal_thread(void* func_ptr)
+{
+    if(!func_ptr) return false;
+
+    Dl_info info;
+    if(dladdr(func_ptr, &info) == 0 || info.dli_fname == nullptr) return false;
+
+    const char* lib_name = info.dli_fname;
+
+    if(std::strstr(lib_name, "libhsa-runtime64") != nullptr) return true;
+    if(std::strstr(lib_name, "librocprofiler-sdk") != nullptr) return true;
+    if(std::strstr(lib_name, "libamdhip64") != nullptr) return true;
+
+    return false;
```

Comment:
Replace the C style `strstr` with C++` str.find`. Don't forget to remove included `cstring` header
```suggestion
    std::string_view lib_name{info.dli_fname};
    
    return lib_name.find("libhsa-runtime64") != std::string::npos ||
           lib_name.find("libhsa-runtime64") != std::string::npos ||
           lib_name.find("libhsa-runtime64") != std::string::npos;
```

Reply:
Done